diff --git a/CMakeLists.txt b/CMakeLists.txt
index 352542a..9a84d02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -271,12 +271,9 @@ find_package(MatrixClient 0.1.0 REQUIRED)
 find_package(Olm 2 REQUIRED)
 find_package(spdlog 1.0.0 CONFIG REQUIRED)
 find_package(cmark REQUIRED)
-find_package(nlohmann_json 3.2.0)
-set_package_properties(nlohmann_json PROPERTIES
-    DESCRIPTION "JSON for Modern C++, a C++11 header-only JSON class"
-    URL "https://nlohmann.github.io/json/"
-    TYPE REQUIRED
-)
+
+pkg_search_module(JSON REQUIRED nlohmann_json>=3.2.0)
+
 
 if(NOT LMDBXX_INCLUDE_DIR)
     find_path(LMDBXX_INCLUDE_DIR
@@ -301,6 +298,8 @@ include_directories(SYSTEM ${TWEENY_INCLUDE_DIR})
 include_directories(${CMAKE_SOURCE_DIR}/src)
 include_directories(${Boost_INCLUDE_DIRS})
 
+include_directories(${JSON_INCLUDE_DIRS})
+
 # local inclue directory
 include_directories(includes)
 
@@ -398,7 +397,7 @@ set(COMMON_LIBS
     Qt5::Svg
     Qt5::Concurrent
     Qt5::Multimedia
-    nlohmann_json::nlohmann_json)
+    ${JSON_LIBRARIES})
 
 if(APPVEYOR_BUILD)
     set(NHEKO_LIBS ${COMMON_LIBS} lmdb)
